lib/kits/core-ui/components/helpers/component.js

```diff
@@ -123,3 +123,25 @@ export const getColorBrightness = (color) => {
 
     return sum <= 381 ? 'dark' : 'light';
 };
+
+export const pushQueryParam = (queryParams = [], action = 'set') => {
+    let urlParams = new URLSearchParams(window.location.search);
+
+    queryParams?.forEach(({name, val}) => {
+        urlParams[action](name, val);
+    });
+
+    if (history.pushState) {
+        urlParams = urlParams.toString() === '' ? '' : `?${urlParams}`;
+        const newUrl = `${window.location.protocol}//${window.location.host}${window.location.pathname}${urlParams}`;
+        window.history.pushState({path: newUrl}, '', newUrl);
+    }
```

Comment:
You can save a lot of this work just by cloning the current URL and mutating it:

```js
export const pushQueryParam = (queryParams = [], action = "set") => {
  const newUrl = new URL(window.location.href);

  queryParams?.forEach(({ name, val }) => {
    newUrl.searchParams[action](name, val);
  });

  window.history.pushState({ path: String(newUrl) }, "", newUrl);
};
```

Also pushState has been around since IE10, I wouldn't bother feature detecting it.